mods/persistence/modules/trade_beacons/cargo_telepads.dm

```diff
@@ -0,0 +1,28 @@
+/obj/machinery/telepad_cargo
+	name = "cargo telepad"
+	desc = "A telepad used to recieve imports and send exports."
+	icon = 'icons/obj/machines/telepad.dmi'
+	icon_state = "pad-idle"
+	anchored = 1
+	use_power = 1
+	idle_power_usage = 20
+	active_power_usage = 500
+	var/telepad_id = 0
+/obj/machinery/telepad_cargo/New()
+	telepad_id = random_id(type,10000,99999)
```

Comment:
Don't override New on atoms. z mimic and other stuff breaks if you do. Put that in Initialize instead.